adlfs/tests/test_spec.py

```diff
@@ -2045,3 +2047,37 @@ def test_open_file_x(storage: azure.storage.blob.BlobServiceClient, tmpdir):
         with fs.open("data/afile", "xb") as f:
             pass
     assert fs.cat_file("data/afile") == b"data"
+
+
+@pytest.mark.parametrize("blocksize", [5 * 2**20, 50 * 2**20, 100 * 2**20])
+def test_number_of_blocks(storage, mocker, blocksize):
+
+    fs = AzureBlobFileSystem(
+        account_name=storage.account_name,
+        connection_string=CONN_STR,
+        blocksize=blocksize,
+    )
+
+    content = b"1" * (blocksize * 2 + 1)
+    with fs.open("data/root/a/file.txt", "wb", blocksize=blocksize) as f:
+        mocker.patch(
+            "azure.storage.blob.aio.BlobClient.commit_block_list", autospec=True
+        )
+        with patch(
+            "azure.storage.blob.aio.BlobClient.stage_block", autospec=True
+        ) as mock_stage_block:
+            f.write(content)
+            expected_blocks = math.ceil(len(content) / blocksize)
+            actual_blocks = mock_stage_block.call_count
+            assert actual_blocks == expected_blocks
+
+
+def test_block_size(storage):
+    fs = AzureBlobFileSystem(
+        account_name=storage.account_name,
+        connection_string=CONN_STR,
+        blocksize=5 * 2**20,
+    )
+
+    with fs.open("data/root/a/file.txt", "wb") as f:
+        assert f.blocksize == 5 * 2**20
```

Comment:
We should probably add an additional parameterized test case where we set the `block_size` directly using the `AzureBlobFile` constructor and have cases that:
1. Assert the default block size for both when `block_size` is ommitted and `block_size` is `None`.
2. Assert block size is **not** inherited from the passed in `fs`. This one may make sense to implement as a separate test method in case the implementation is not clean as part of parameterization
3. Assert that we can override `block_size`